src/sqe.rs

```diff
@@ -159,40 +138,22 @@ impl<'a> SubmissionQueueEvent<'a> {
     ) {
         let len = buf.len();
         let addr = buf.as_ptr();
-        self.sqe.opcode = IORING_OP_WRITE_FIXED as _;
-        self.sqe.fd = fd;
-        self.sqe.off_addr2.off = offset as _;
-        self.sqe.addr = addr as _;
-        self.sqe.len = len as _;
-        self.sqe.buf_index.buf_index = buf_index as _;
+        sys::io_uring_prep_write_fixed(self.sqe, fd, addr as _, len as _, offset as _, buf_index as _);
     }
 
     #[inline]
     pub unsafe fn prep_fsync(&mut self, fd: RawFd, flags: FsyncFlags) {
-        self.sqe.opcode = IORING_OP_FSYNC as _;
-        self.sqe.fd = fd;
-        self.sqe.off_addr2.off = 0;
-        self.sqe.addr = 0;
-        self.sqe.len = 0;
-        self.sqe.cmd_flags.fsync_flags = flags.bits() as _;
+        sys::io_uring_prep_fsync(self.sqe, fd, flags.bits() as _);
     }
 
     #[inline]
-    pub unsafe fn prep_timeout(&mut self, ts: &sys::__kernel_timespec) {
-        self.sqe.opcode = IORING_OP_TIMEOUT as _;
-        self.sqe.fd = 0;
-        self.sqe.addr = ts as *const sys::__kernel_timespec as _;
-        self.sqe.len = 1;
-        self.sqe.user_data = sys::LIBURING_UDATA_TIMEOUT;
+    pub unsafe fn prep_timeout(&mut self, ts: &sys::__kernel_timespec, count: usize, flags: TimeoutFlags) {
```

Comment:
Good catch on the count parameter :smile: 

What do you think about splitting this function into one just for `__kernel_timespec` timeouts and one like this with both count and timeout? That way users who only care about time don't have to worry about what `count = 0` entails.

Reply:
I think that's a great idea (and I actually have not looked up what it means to set count to a nonzero value! all I know is that it sets the offset field to this value). All we need is a set of names.

Reply:
It's also unfortunate that we have to expose the `__kernel_timespec` type, but since its passed as an address the address must remain live until after the `submit.` Otherwise I'd just take a `Duration` here and construct a `__kernel_timespec` in the function.

(I had the same problem with trying to create `prep_read` and `prep_write` helpers, which cannot construct the slice of iovecs because then they'll be invalidated before the submit.)

Reply:
> I think that's a great idea (and I actually have not looked up what it means to set count to a nonzero value! all I know is that it sets the offset field to this value. does it repeat the timeout or something?). All we need is a set of names.

I think count is an optional number of completed events. Then the timeout completes based on either the timeout firing or completing that number of events (excluding other timeouts I believe). The `res` field based on count events is `0` (no error), but timeout events will be `-ETIME`. 

I am basing this off this nice example: 
https://github.com/axboe/liburing/blob/556f71e5ec522ea46ed1e7a4a9a31886fd00b364/test/timeout.c#L610

Reply:
>It's also unfortunate that we have to expose the __kernel_timespec type, but since its passed as an address the address must remain live until after the submit. Otherwise I'd just take a Duration here and construct a __kernel_timespec in the function.

It might be needlessly complex, but maybe we could introduce some sort of inner `'submit` lifetime? I don't really know what it would mean to call `submit` functions then, or how to actually implement it 😅

Reply:
In the higher level library that I haven't yet published, I create a `Submitter` type which also contains a `&mut [libc::iovec; 1]`, so that it can have a simple `prep_read`. That field could also be used for timeouts taking a duration if we make the element of the array a union of `iovec` and `kernel_timespec`. 

The lifetime of the reference in the field must outlive the submit call or it is incorrect, but the library at the next level up is responsible for making sure that happens.